modules/setting/user.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package setting
+
+import (
+	"strings"
+
+	"code.gitea.io/gitea/modules/log"
+)
+
+// userSetting represents user settings
+type userSetting struct {
+	SettingDisabledModules []string
+}
+
+func (s *userSetting) Enabled(module string) bool {
+	for _, m := range s.SettingDisabledModules {
+		if strings.EqualFold(m, module) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
Could use a `Set` with lowercase keys. This would enable a fast lookup and the pre-processing of the keys would prevent the constant case checks. On the other hand, the used routes may not be performance critical.